papaya-web/src/main/java/pl/edu/agh/papaya/rest/common/UserIdentificationService.java

```diff
@@ -0,0 +1,34 @@
+package pl.edu.agh.papaya.rest.common;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.edu.agh.papaya.api.model.UserIdentificationDto;
+import pl.edu.agh.papaya.model.User;
+import pl.edu.agh.papaya.service.user.UserService;
+
+@Component
+public class UserIdentificationService {
+
+    @Autowired
+    private UserService userService;
+
+    public Optional<User> identify(UserIdentificationDto userIdentification) {
+        Long userId = userIdentification.getId();
+        String userEmail = userIdentification.getEmail();
+
+        if (userId != null && userEmail != null) {
```

Comment:
To specjalnie, tak? Jedno chcemy puste? Nie jest to intuicyjny warunek na pierwszy rzut oka :stuck_out_tongue:

Reply:
Każde z tych dwóch pól jednoznacznie identyfikuje usera więc możemy:
1. wymagać aby tylko jedno było ustawione,
2. jeśli oba są ustawione sprawdzać czy się zgadzają.

Reply:
No wiem, tylko się upewniam, że dobrze rozumiem intencje - to drugie podejście nie brzmi dla mnie źle - jako użytkownik API nie zakładałabym, że muszę podać dokładnie jedno z tych pól (brzmi jak coś z dokumentacji C)

Reply:
Poprawione